src/lib/integration_test_local_engine/mina_docker.ml

```diff
@@ -0,0 +1,496 @@
+open Core
+open Async
+open Currency
+open Signature_lib
+open Mina_base
+open Integration_test_lib
+
+let docker_swarm_version = "3.9"
+
+module Network_config = struct
+  module Cli_inputs = Cli_inputs
+
+  type docker_volume_configs = { name : string; data : string }
+  [@@deriving to_yojson]
+
+  type block_producer_config =
+    { name : string
+    ; id : string
+    ; keypair : Network_keypair.t
+    ; public_key : string
+    ; private_key : string
+    ; keypair_secret : string
+    ; libp2p_secret : string
+    }
+  [@@deriving to_yojson]
+
+  type snark_coordinator_configs =
+    { name : string
+    ; id : string
+    ; public_key : string
+    ; snark_worker_fee : string
+    }
+  [@@deriving to_yojson]
+
+  type docker_config =
+    { docker_swarm_version : string
+    ; stack_name : string
+    ; coda_image : string
+    ; docker_volume_configs : docker_volume_configs list
+    ; block_producer_configs : block_producer_config list
+    ; snark_coordinator_configs : snark_coordinator_configs list
+    ; log_precomputed_blocks : bool
+    ; archive_node_count : int
+    ; mina_archive_schema : string
+    ; runtime_config : Yojson.Safe.t
+          [@to_yojson fun j -> `String (Yojson.Safe.to_string j)]
+    }
+  [@@deriving to_yojson]
+
+  type t =
+    { keypairs : Network_keypair.t list
+    ; debug_arg : bool
+    ; constants : Test_config.constants
+    ; docker : docker_config
+    }
+  [@@deriving to_yojson]
+
+  let expand ~logger ~test_name ~cli_inputs:_ ~(debug : bool)
```

Comment:
a large amount of this func doesn't look implementation specific, i feel like we should find a place in the /integration_test_lib dir for many of the pieces.  keys, block producer configs, constants, names of things, etc.  probably shoulda done this to begin with haha.  if we wanna just merge this quicker then no worries but would be nice to rationalize it a bit more, if not now then later for sure

Reply:
Totally, Nathan mentioned refactoring a lot of the test setup steps into something more reusable. I didn't go down that route just because I didn't want to be refactoring the cloud engine at the same time in this PR. This will be an easy enough change in a separate PR as long as we keep the new interface consistent.